gui/src/components/usagesWarning.tsx

```diff
@@ -0,0 +1,133 @@
+import { useState, useEffect } from 'react';
+import { AlertTriangle, X } from 'lucide-react';
+import { SERVER_URL } from "../../../core/util/parameters";
+import * as vscode from 'vscode';
+
+export default function UsageWarning() {
+  const [isVisible, setIsVisible] = useState(false);
+  const [message, setMessage] = useState("");
+
+  const token = localStorage.getItem('userToken') || "";
+  const url = `${SERVER_URL}/get-usage`;
+
+  const getMessage = async () => {
+    try {
+      const res = await fetch(url, {
+        method: 'GET',
+        headers: {
+          "Content-Type": "application/json",
+          Authorization: `Bearer ${token}`,
+        }
+      });
+
+      if (!res.ok) {
+        console.log('Error fetching usage data');
+        return;
+      }
+
+      const { warning_message } = await res.json();
+      if (warning_message) {
+        setMessage(warning_message);
+        setIsVisible(true); 
+        vscode.window.showWarningMessage(warning_message);
+      } else {
+        setMessage("");
+        setIsVisible(false); 
+      }
+    } catch (error) {
+      console.log('Error fetching usage data:', error);
+    }
+  };
+
+  useEffect(() => {
+    getMessage();
+  }, []);
```

Comment:
this will only fetch when component gets mounted, so if users is overusing credits it in a single go in one session, he wouldn't get notified right.

Reply:
Thinking of using setInterval to poll every 5 minutes for credit usage—open to better ideas if anyone has them!